stage1/init/init.go

```diff
@@ -97,6 +99,39 @@ func init() {
 	runtime.LockOSThread()
 }
 
+// machinedRegister checks if nspawn should register the pod to machined
+func machinedRegister() bool {
+	// machined has a D-Bus interface following versioning guidelines, see:
+	// http://www.freedesktop.org/wiki/Software/systemd/machined/
+	// Therefore we can just check if the D-Bus method we need exists and we
+	// don't need to check the signature.
+	var found int
+
+	conn, err := dbus.SystemBus()
+	if err != nil {
+		return false
+	}
+	node, err := introspect.Call(conn.Object("org.freedesktop.machine1", "/org/freedesktop/machine1"))
+	if err != nil {
+		return false
+	}
+	for _, iface := range node.Interfaces {
+		if iface.Name != "org.freedesktop.machine1.Manager" {
+			continue
+		}
+		for _, method := range iface.Methods {
+			if method.Name == "CreateMachineWithNetwork" || method.Name == "RegisterMachineWithNetwork" {
+				found++
+			}
+		}
+		if found == 2 {
+			return true
+		}
```

Comment:
If found != 2 here we've found "org.freedesktop.machine1.Manager" but not the desired methods, and we'll continue iterating on `node.Interfaces` which doesn't seem useful.

Why not break unconditionally after `for _, method := range iface.Methods { }`, then test found after `for _, iface := range node.Interfaces { }` to determine the return value?